ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/SandboxIndexAndWriteService.scala

```diff
@@ -326,11 +327,13 @@ private class SandboxIndexAndWriteService(
       case LedgerOffset.Absolute(absBegin) =>
         ledger.ledgerEntries(Some(absBegin.toLong)).collect {
           case (offset, t: LedgerEntry.Transaction)
-              if (t.applicationId == applicationId.unwrap && parties.contains(t.submittingParty)) =>
+              if t.applicationId.contains(applicationId.unwrap) &&
+                t.submittingParty.exists(parties.contains) &&
+                t.commandId.nonEmpty =>
```

Comment:
Why the check for existing command ID?

Reply:
Because previously it wasn't optional, and to be able to emit `CommandAccepted` we need the commandId.

Reply:
Ah, and I assume the index service should not emit completions for non-local transactions (i.e., those with missing command IDs)?

Reply:
This change makes more sense in a future commit where this code is moved to a generic `LedgerBackedIndexService` which in turn will be used by the ApiIndexServer and the sandbox.
In the sandbox this doesn't make a difference, but for the ApiIndexServer, we should only emit `CommandAccepted` for transactions that have "submitter" information (appId, submitter, commandId), meaning the transaction originated from this api server and there is a client listening for these completions.

Thanks for asking the question. I added a comment in the code as well.

Reply:
:+1: